modules/client-common/test/unit/encoding.test.ts

```diff
@@ -0,0 +1,452 @@
+          ]),
+          skip0x: true,
+          output: "0a141e28323c46505a64c8faff",
+        },
+        {
+          buffer: new Uint8Array([100, 100, 100, 100, 100, 100]),
+          skip0x: false,
+          output: "0x646464646464",
+        },
+        {
+          buffer: new Uint8Array([100, 100, 100, 100, 100, 100]),
+          skip0x: true,
+          output: "646464646464",
+        },
+        { buffer: new Uint8Array([0, 255]), skip0x: false, output: "0x00ff" },
+        { buffer: new Uint8Array([0, 255]), skip0x: true, output: "00ff" },
+      ];
+
+      for (let item of items) {
+        const hex = bytesToHex(item.buffer, item.skip0x);
+        expect(hex).toEqual(item.output);
+      }
+    });
+  });
+  describe("encodeRatio", () => {
+    it("Should encode a bigint from a float and the number of digits", () => {
+      const inputs = [
+        { float: .5, digits: 1, out: BigInt(5) },
+        { float: 1, digits: 4, out: BigInt(10000) },
+        { float: 0.25555, digits: 2, out: BigInt(26) },
+        { float: 0.123456789, digits: 15, out: BigInt(123456789000000) },
+      ];
+
+      for (let input of inputs) {
+        const result = encodeRatio(input.float, input.digits);
+        expect(result.toString()).toEqual(input.out.toString());
+      }
+
+      expect(() => encodeRatio(0.5, -1)).toThrow(
+        "The number of digits should range between 1 and 15",
+      );
+      expect(() => encodeRatio(0.5, 18)).toThrow(
+        "The number of digits should range between 1 and 15",
+      );
+    });
+  });
+  describe("decodeRatio", () => {
+    it("Should decode a float from a given bigint and a number of digits", () => {
+      const inputs = [
+        { bigint: BigInt(5), digits: 1, out: 0.5 },
+        { bigint: BigInt(5456), digits: 4, out: 0.5456 },
+        { bigint: 5, digits: 1, out: 0.5 },
+        { bigint: 5456, digits: 4, out: 0.5456 },
+        { bigint: BigInt("1"), digits: 9, out: 0.000000001 },
+        { bigint: BigInt("367483947"), digits: 9, out: 0.367483947 },
+        { bigint: 1, digits: 9, out: 0.000000001 },
+        { bigint: 367483947, digits: 9, out: 0.367483947 },
+      ];
+
+      for (let input of inputs) {
+        const result = decodeRatio(input.bigint, input.digits);
+        expect(result).toEqual(input.out);
+      }
+
+      expect(() => decodeRatio(Number.MAX_SAFE_INTEGER + 1, 12)).toThrow(
+        "The value is out of range",
+      );
+      expect(() => decodeRatio(BigInt("512345898367483947"), 12)).toThrow(
+        "The value is out of range",
+      );
+      expect(() => decodeRatio(10 ** 2, 1)).toThrow(
+        "The value is out of range",
+      );
+      expect(() => decodeRatio(10 ** 10, 9)).toThrow(
+        "The value is out of range",
+      );
+    });
+  });
+  describe("encodeProposalId", () => {
+    it("Should encode a nonce-based proposalId", () => {
+      const entries = [
+        {
+          addr: "0x0000000000000000000000000000000000000000",
```

Comment:
In OSx, we use format string for better readability. E.g.,
```ts
`0x${'0'.repeat(39)}1`, // address(1)
```
Maybe you want to use this too

Reply:
Mmm, i thing is going to be a bit messier to change it just here and leave in the other places as a harcoded string an just here as a function with, repeat, I can considerchanging it But I'd do it in all places at the same time